lib/ansible/cli/doc.py

```diff
@@ -1358,7 +1358,7 @@ def get_role_man_text(self, role, role_json):
                     text.append(doc.pop('examples').strip())
                 else:
                     try:
-                        text.append(yaml_dump(doc.pop('examples'), indent=2, default_flow_style=False))
+                        text.append(DocCLI._dump_yaml(doc.pop('examples')))
```

Comment:
Is there a reason we're using the AnsibleLoader and AnsibleDumper instead of the normal loader and dumper?

Reply:
`!vault` and  `!unsafe` in examples break otherwise

Reply:
@mattclay @sivel ping

Reply:
There doesn't appear to be a good reason to add support to ansible-doc for dumping Ansible-specific types. Switching to the standard YAML loader for sidecar will avoid the need for that.

Looking at collections on Galaxy, there's no usage of Ansible YAML tags in sidecar docs. Even if someone was using them, the docs are already broken -- either generating tracebacks or rendering incorrectly. So we shouldn't need to worry about introducing new issues by changing the loader.

Reply:
This is not a new feature but fixing the exiting feature to align with what would be the logical  expectation, that ansible documentation can handle ansible specific yaml. 

Also, I'm sure no one is using a feature that would currently break their docs, that does not mean they have not tried nor want to.